lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -203,7 +203,17 @@ defmodule Mix.Tasks.Escript.Build do
         []
       end
 
-    Enum.filter(extra_apps, &(&1 in [:eex, :ex_unit, :mix, :iex, :logger]))
+    apps_from_deps = Enum.flat_map(Mix.Dep.cached(), &apps_for_dep/1)
```

Comment:
I don't think we should do this but rather do the same we would do in a release: i.e. find the app for current app and then traverse all applications from there. :)